src/core/cgroup.c

```diff
@@ -1848,14 +2028,19 @@ static int unit_realize_cgroup_now(Unit *u, ManagerState state) {
         if (unit_has_mask_realized(u, target_mask, enable_mask))
                 return 0;
 
-        /* First, realize parents */
+        /* Disable controllers below us, if there are any */
+        r = unit_realize_cgroup_now_disable(u, state);
```

Comment:
hmm, interesting, if you do this in two explicit steps now, then i figure we don't need 27adcc97377 anymore, do we?

Reply:
Hmm, probably not. I kind of prefer the explicit route since it's easy to reason about, and matches up with the existing slice check logic. What do you think?